dp-cluster-setup-utility.py

```diff
@@ -1213,67 +1616,216 @@ def satisfied(self):
+"""
+class CMOperationsManager(BaseOperationsManager):
+  def __init__(self):
+    pass
+  @staticmethod
+  def get_instance():
+    return ClouderaManager(user.url_input('CM URL', 'cm.url'), user.credential_input('CM admin', 'cm.admin'))
+  
+  @staticmethod
+  def pre_reqs_satisfied(cm):
+    return CMPrerequisites(cm).satisfied()
+  
+  @staticmethod
+  def get_topology_util(cm, dp_dependencies):
+    return CMTopologyUtil(cm, dp_dependencies)
+  
+  @staticmethod
+  def get_knox_details(cm):
+    print("Skipping knox input data for CM based cluster")
+    pass
 
-  if 'BEACON' in dp.dependency_names():
-      topologies_to_deploy.extend([BeaconProxyTopology(ambari, dp.dependency_names(), topology_util)])
+  @staticmethod
+  def get_topologies_to_deploy(dp, cm, topology_util):
+    print("Skipping topology deployment for CM based cluster")
+    return []
 
-  if 'DATA_ANALYTICS_STUDIO' in dp.dependency_names():
-    topologies_to_deploy.extend([TokenTopology(dp.public_key(), 'redirecttoken', 10000), RedirectTopology('redirect')])
-  for topology in topologies_to_deploy:
-    print 'Deploying Knox topology:', topology.name
-    topology.deploy(knox)
+  @staticmethod
+  def check_cluster_for_dp(dp,knox):
+    return dp.check_cm(knox)
 
-  if 'RANGER' in dp.dependency_names() or dp.optional_dependency_names():
-    ambari.enable_trusted_proxy_for_ranger()
-  if 'ATLAS' in dp.dependency_names() or dp.optional_dependency_names():
-    ambari.enable_trusted_proxy_for_atlas()
-  if 'BEACON' in dp.dependency_names() or dp.optional_dependency_names():
-    ambari.enable_trusted_proxy_for_beacon()
+  @staticmethod
+  def register_cluster_with_dp(dp, cm, knox, user):
+    return dp.register_cm(cm, user)
 
-  print 'Waiting for knox topologies to get activated. Sleeping for 10 seconds...'
-  time.sleep(10)
-  print 'Cluster changes are complete! Please log into Ambari, confirm the changes made to your cluster as part of this script and restart affected services.'
-  user.any_input()
+"""
+  Execution Starts here
+"""  
+if __name__ == '__main__':
+  user = Memorized(User())
+  print '\nThis script will check to ensure that all necessary pre-requisites have been met and then register this cluster with DataPlane.\n'
+  print 'Please ensure that your cluster has kerberos enabled, Ambari has been configured to use kerberos for authentication, and Knox is installed. Once those steps have been done, run this script from the Knox host and follow the steps and prompts to complete the cluster registration process.\n'
 
-  if not dp.check_ambari(knox):
+  if not ScriptPrerequisites().satisfied():
     sys.exit(1)
-
-  print 'Registering cluster to DataPlane...'
-  response = dp.register_ambari(ambari, knox, user)
-  print 'Cluster is registered with id', response['id']
-
-  print 'Success! You are all set, your cluster is registered and ready to use.'
+  # Get the cluster type and execute the flow
+  print('Tell me about your Cluster type')
```

Comment:
I think we should provide options here like 'Tell me about your Cluster type (HDP/HDF/CDH)' and then accept one of these, based on this, we can map the cluster manager internally to Ambari or CM. What do you think?

Reply:
We can do that

Reply:
Can you please change the message as indicated? It will not be clear to the user what to enter without a prompt.

Reply:
The message shown to user is 
```bash
Tell me about your Cluster type
Cluster Type  [HDP/HDF/CDH] [CDH]:
```